test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/FullUriFunctionalTests.cs

```diff
@@ -71,52 +71,63 @@ public void SelectOrExpandCanOnlyBeCalledOnEntity()
             parseWithNonEntity.ShouldThrow<ODataException>().WithMessage(ODataErrorStrings.UriParser_TypeInvalidForSelectExpand("Edm.String"));
         }
 
-        [Fact(Skip = "This test currently fails.")]
+        [Fact]
         public void ParseSelectExpand()
         {
             ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData"), new Uri("http://www.odata.com/OData/Dogs?$select=Color, MyPeople&$expand=MyPeople"));
             ODataUri parsedUri = parser.ParseUri();
-            parsedUri.SelectAndExpand.SelectedItems.First().ShouldBePathSelectionItem(new ODataPath(new PropertySegment(HardCodedTestModel.GetDogColorProp())));
-            var myPeopleExpand = parsedUri.SelectAndExpand.SelectedItems.Last().ShouldBeSelectedItemOfType<ExpandedNavigationSelectItem>().And;
+            parsedUri.SelectAndExpand.SelectedItems.Count().Should().Be(3);
+            parsedUri.SelectAndExpand.SelectedItems.ElementAt(1).ShouldBePathSelectionItem(new ODataPath(new PropertySegment(HardCodedTestModel.GetDogColorProp())));
+            var myPeopleExpand = parsedUri.SelectAndExpand.SelectedItems.First().ShouldBeSelectedItemOfType<ExpandedNavigationSelectItem>().And;
             myPeopleExpand.PathToNavigationProperty.Single().ShouldBeNavigationPropertySegment(HardCodedTestModel.GetDogMyPeopleNavProp());
             var myPeopleSelectExpand = myPeopleExpand.SelectAndExpand;
             myPeopleSelectExpand.AllSelected.Should().BeTrue();
             myPeopleSelectExpand.SelectedItems.Should().BeEmpty();
         }
 
-        [Fact(Skip = "This test currently fails.")]
+        [Fact]
         public void ParseTop()
         {
             ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData"), new Uri("http://www.odata.com/OData/People?$top=1"));
             ODataUri parsedUri = parser.ParseUri();
             parsedUri.Top.Should().Be(1);
         }
 
-        [Fact(Skip = "This test currently fails.")]
+        [Fact]
         public void TopOnlyValidOnEntities()
         {
             ODataUriParser parser = new ODataUriParser(HardCodedTestModel.TestModel, new Uri("http://www.odata.com/OData"), new Uri("http://www.odata.com/OData/People(1)/MyAddress?$top=1"));
-            Action parseWitnNonEntity = () => parser.ParseUri();
-            parseWitnNonEntity.ShouldThrow<ODataException>().WithMessage(ODataErrorStrings.MetadataBinder_QueryOptionNotApplicable("$top"));
+            ODataUri parsedUri = parser.ParseUri();
+            parsedUri.Top.Should().Be(1);

```

Comment:
Assuming the Uri is valid, should we update the method test name?
q: why it verification is on parsedUri.Top? Can it be something else more specific from the parsed result?